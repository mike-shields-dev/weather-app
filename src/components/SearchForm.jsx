import React from "react";
import PropTypes from "prop-types";
import styles from "../styles/SearchForm.module.css";

const SearchForm = ({
  handleChange,
  handleSubmit,
  searchText,
  placeholderText,
}) => {
  return (
    <div className="search-form" data-testid="search-form">
      <input
        className={`${styles["search-form__input"]}`}
        data-testid="search-form-input"
        onChange={handleChange}
        type="text"
        placeholder={placeholderText}
        value={searchText}
      />
      <button
        className="search-form__button shadow"
        data-testid="search-form-button"
        onClick={handleSubmit}
        type="submit"
      >
        search
      </button>
    </div>
  );
};

SearchForm.defaultProps = {
  placeholderText: "Enter your search",
  searchText: "",
};

SearchForm.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  searchText: PropTypes.string,
  placeholderText: PropTypes.string,
};

export default SearchForm;
