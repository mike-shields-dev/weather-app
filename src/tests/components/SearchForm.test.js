import React from "react";
import { fireEvent, render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

const validProps = {
  handleChange: jest.fn(),
  handleSubmit: jest.fn(),
  placeholderText: "Enter city name",
};

describe("SearchForm", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<SearchForm {...validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a search text input", () => {
    const { getByRole } = render(<SearchForm {...validProps} />);

    expect(getByRole("textbox")).toBeInTheDocument();
  });

  it("search text input renders provided placeholder text", () => {
    const { getByTestId } = render(<SearchForm {...validProps} />);
    const searchInput = getByTestId("search-form-input");

    expect(searchInput.placeholder).toEqual(validProps.placeholderText);
  });

  it("search text input renders default placeholder if no placeholder text prop is provided", () => {
    const { placeholderText, ...validPropsNoPlaceholderText } = validProps;
    const { getByTestId } = render(
      <SearchForm {...validPropsNoPlaceholderText} />
    );
    const searchInput = getByTestId("search-form-input");

    expect(searchInput.placeholder).toEqual("Enter your search");
  });

  it("typing into the search text input calls handleChange function", async () => {
    const { getByRole } = render(<SearchForm {...validProps} />);
    const textInput = getByRole("textbox");

    fireEvent.change(textInput, { target: { value: "Manchester" } });

    expect(validProps.handleChange).toHaveBeenCalled();
  });

  it("renders a submit button", () => {
    const { getByRole } = render(<SearchForm {...validProps} />);

    expect(getByRole("button", { type: "submit" })).toBeInTheDocument();
  });

  it("clicking submit button calls handleSubmit function", () => {
    const { getByRole } = render(<SearchForm {...validProps} />);
    const button = getByRole("button");

    fireEvent.click(button);

    expect(validProps.handleSubmit).toHaveBeenCalled();
  });
});
