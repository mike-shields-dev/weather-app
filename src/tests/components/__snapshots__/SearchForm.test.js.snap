// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="search-form"
    data-testid="search-form"
  >
    <input
      class="search-form__input"
      data-testid="search-form-input"
      placeholder="Enter city name"
      type="text"
      value=""
    />
    <button
      class="search-form__button shadow"
      data-testid="search-form-button"
      type="submit"
    >
      search
    </button>
  </div>
</DocumentFragment>
`;
